src/python/pants/cache/cache_setup.py

```diff
@@ -361,6 +367,7 @@ def create_local_cache(parent_path):
             )
             return LocalArtifactCache(
                 artifact_root,
+                artifact_extraction_root,
```

Comment:
Thanks, Henry! The implementation works, although I was wondering if you had a chance to try avoiding adding the param `artifact_extraction_root`, and instead directly replace `artifact_root` with `artifact_extraction_root`. Does that break anything?

Reply:
Maybe I'm misunderstanding but wouldn't that be the same as what was done in #8302 that led to the revert?

Reply:
You are correct. Sorry I should've mentioned this earlier:

The issue was likely at https://github.com/pantsbuild/pants/blob/f8d7f28228bccb10bb20497607875e1d6ba4a45d/src/python/pants/cache/artifact.py#L110

E.g. with `pants-physical-workdir-base: /tmp/123`:
`path = /Users/user/workspace/pants/.pants.d/...` and 
`self._artifact_root = /tmp/123/.pants_123/`, which makes 
`relpath = ../../../../tmp/123/.pants_123` thus causing bad cache entry.

One possibility is to see whether we can limit `paths` to be relative from the `workdir` to start with. I trust your judgment on this, so please feel free to merge if this is still the more reasonable approach.

Thanks!

Reply:
Just for posterity let me see if I can defend my approach.
> One possibility is to see whether we can limit paths to be relative from the workdir to start with

In the line you referenced the "path" variable contains a symlink to the pants workdir. If we wanted to use the readlinked wordir when encoding the tar we would have to do `relpath = os.path.relpath(readlink(path), self._artifact_extraction_root)` in order to get the relpath correct. I think that would also work to get the paths in the tar correct, but adds O(n) extra readlinks when we are creating artifacts, where n is the number of output files we created. I think your method is a little more elegant because you don't need the extra argument, but it increased the number of readlinks we have to do which is what we're trying to prevent.

Reply:
Thanks for the thorough feedback btw 😄